octoprint_homeassistant/__init__.py

```diff
@@ -659,27 +659,35 @@ def _generate_psu_state(self, psu_state=None):
     def _on_emergency_stop(
         self, topic, message, retained=None, qos=None, *args, **kwargs
     ):
+        # In Home Assistant, MQTT buttons send the message 'PRESS' when pressed.
         self._logger.debug("Emergency stop message received: " + str(message))
-        if message:
+        if message == b"PRESS":
             self._printer.commands("M112")
+        else:
+            self._logger.error("Unknown message received: " + str(message))
 
     def _on_cancel_print(
         self, topic, message, retained=None, qos=None, *args, **kwargs
     ):
         self._logger.debug("Cancel print message received: " + str(message))
-        if message:
+        if message == b"PRESS":
             self._printer.cancel_print()
+        else:
+            self._logger.error("Unknown message received: " + str(message))
 
     def _on_pause_print(self, topic, message, retained=None, qos=None, *args, **kwargs):
```

Comment:
This PR was originally meant to just fix #96. When a switch is toggled in Lovelace, the inverted state of the switch is sent as a payload to the MQTT topic. If the switch is on Lovelace sends `False`, and if the switch is off Lovelace sends `True`. Additionally, when a button is pressed in Lovelace, the string `PRESS` is sent. The only necessary changes are this one here and in the `_on_connect_printer()` function. The checks for `PRESS` and/or the check for the presence of a message in general everywhere else is technically unnecessary.